Tests/MessagingInApp/Views/InAppMessageViewTest.swift

```diff
@@ -658,6 +659,157 @@ class InAppMessageViewTest: IntegrationTest {
+
+        let givenInlineMessage1 = Message.randomInline
+        let givenInlineMessage2 = Message.randomInline
+        queueMock.getInlineMessagesReturnValue = [givenInlineMessage1, givenInlineMessage2]
+
+        let inlineView1 = InAppMessageView(elementId: givenInlineMessage1.elementId!)
+        let inlineView2 = InAppMessageView(elementId: givenInlineMessage2.elementId!)
+
+        // Render only inlineView1
+        await onDoneRenderingInAppMessage(givenInlineMessage1, insideOfInlineView: inlineView1)
+
+        // inlineView1 is only visible
+        XCTAssertTrue(isInlineViewVisible(inlineView1))
+        XCTAssertFalse(isInlineViewVisible(inlineView2))
+
+        // Check if messageShown is called
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 1)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 1)
+
+        // Dismiss inlineView1 and show inlineView2
+        await onCloseActionButtonPressed(onInlineView: inlineView1)
+        await onDoneRenderingInAppMessage(givenInlineMessage2, insideOfInlineView: inlineView2)
+
+        // Check messageShownCalled called again
+        XCTAssertTrue(eventListenerMock.messageShownCalled)
+        XCTAssertEqual(eventListenerMock.messageShownCallsCount, 2)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 2)
+    }
+
+    // MARK: - Track click
+
+    @MainActor
+    func test_onInlineMessageShown_givenDelegateNotSet_onSingleButtonTap_expectTrackClickedMetric_expectGlobalMessageActionTakenCallback() async {
+        DIGraphShared.shared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)
+        let inlineView = await showInlineMessageForMetrics()
+        XCTAssertTrue(isInlineViewVisible(inlineView))
+
+        // Tap button once
+        onCustomActionButtonPressed(onInlineView: inlineView)
+
+        // Check if messageActionTaken is called
+        XCTAssertTrue(eventListenerMock.messageActionTakenCalled)
+        XCTAssertEqual(eventListenerMock.messageActionTakenCallsCount, 1)
+
+        // Check other handlers are not called
+        XCTAssertFalse(eventListenerMock.messageDismissedCalled)
+        XCTAssertFalse(eventListenerMock.errorWithMessageCalled)
+
+        // Also check for postEvent calls
+        XCTAssertTrue(eventBusHandlerMock.postEventCalled)
+        // The first post call occurs when the message is shown,
+        // and the second post call is triggered by the button click action.
+        XCTAssertEqual(eventBusHandlerMock.postEventCallsCount, 2)
+    }
+
+    @MainActor
+    func test_onInlineMessageShown_givenDelegateNotSet_onMultipleButtonTap_expectGlobalMessageActionTakenCallback() async {
```

Comment:
I think this test should be broken into multiple test functions. I also think event listener tests out of scope for this ticket.

Reply:
messageShown and messageActionTaken cases are handled in this PR as per description [here](https://github.com/customerio/customerio-ios/pull/775)